confdb/confdb_test.go

```diff
@@ -2843,3 +2843,94 @@ func (*viewSuite) TestCheckWriteEphemeralAccess(c *C) {
 		}
 	}
 }
+
+func (*viewSuite) TestContentAccessExamples(c *C) {
+	type testcase struct {
+		rule           map[string]interface{}
+		nestedReadErr  string
+		nestedWriteErr string
+	}
+
+	tcs := []testcase{
+		{
+			rule: map[string]interface{}{
+				"storage": "a",
+				"access":  "read",
+				"content": []interface{}{
+					map[string]interface{}{"storage": "b", "access": "write"},
+				},
+			},
+			nestedReadErr:  `cannot get "a\.b" through acc/foo/my-view: no matching rule`,
+			nestedWriteErr: "",
+		},
+		{
+			rule: map[string]interface{}{
+				"storage": "a",
+				"access":  "write",
+				"content": []interface{}{
+					map[string]interface{}{"storage": "b", "access": "read"},
+				},
+			},
+			nestedReadErr:  "",
+			nestedWriteErr: `cannot set "a\.b" through acc/foo/my-view: no matching rule`,
+		},
+		{
+			// no "inherited" behaviour
+			rule: map[string]interface{}{
+				"storage": "a",
+				"access":  "read",
+				"content": []interface{}{
+					map[string]interface{}{"storage": "b"},
+				},
+			},
+			nestedReadErr:  "",
+			nestedWriteErr: "",
+		},
+		{
+			// no "inherited" behaviour
+			rule: map[string]interface{}{
+				"storage": "a",
+				"access":  "write",
+				"content": []interface{}{
+					map[string]interface{}{"storage": "b"},
+				},
+			},
+			nestedReadErr:  "",
+			nestedWriteErr: "",
+		},
+	}
+
+	for _, tc := range tcs {
+		databag := confdb.NewJSONDatabag()
+		err := databag.Set("a", map[string]interface{}{
+			"b": 123,
+		})
+		c.Assert(err, IsNil)
+
+		schema, err := confdb.NewSchema("acc", "foo", map[string]interface{}{
+			"my-view": map[string]interface{}{
+				"rules": []interface{}{
+					tc.rule,
+				},
+			},
+		}, confdb.NewJSONSchema())
+		c.Assert(err, IsNil)
+
+		view := schema.View("my-view")
+
+		res, err := view.Get(databag, "a.b")
```

Comment:
as discussed we also want to try what happens if read/write via a itself and get the entire map, or set the map  {"b": value}

Reply:
Those aren't affected by the presence of b, they just match with a. I've extended the test to cover reading/writing "a"